src/bin/set-version/set_version.rs

```diff
@@ -132,13 +132,37 @@ fn exec(args: VersionArgs) -> CargoResult<()> {
             .collect::<Vec<_>>()
     };
 
+    let update_workspace_version = all
+        || selected.iter().any(|p| {
+            LocalManifest::try_new(Path::new(&p.manifest_path))
+                .map_or(false, |m| m.version_is_inherited())
+        });
+    if update_workspace_version {
+        let mut ws_manifest = LocalManifest::try_new(&root_manifest_path)?;
+        if let Some(current) = ws_manifest.get_workspace_version() {
+            if let Some(next) = target.bump(&current, metadata.as_deref())? {
+                ws_manifest.set_workspace_version(&next);
+                shell_status(
+                    "Upgrading",
+                    &format!("workspace version from {} to {}", current, next),
+                )?;
+
+                if !dry_run {
+                    ws_manifest.write()?;
+                }
+            }
+        }
+    }
+
     for package in selected {
         let current = &package.version;
         let next = target.bump(current, metadata.as_deref())?;
         if let Some(next) = next {
             {
                 let mut manifest = LocalManifest::try_new(Path::new(&package.manifest_path))?;
-                manifest.set_package_version(&next);
+                if !manifest.version_is_inherited() {
+                    manifest.set_package_version(&next);
+                }
 
                 shell_status(
                     "Upgrading",
```

Comment:
I feel like the `if` check should be moved to this wider block.  We could then provide a tailored `shell_status` to for when the upgrade happened via the workspace.version.

We should also have a comment that is the companion to the workspace update that we still need to run in the inherit mode to update dependents

Reply:
I'm not sure I fully understood what you meant, is my last version ok ?